make/targets/openshift/tests-extension.mk

```diff
@@ -0,0 +1,97 @@
+#   )
+#
+#   TESTS_EXT_BINARY := my-operator-tests-ext
+#   TESTS_EXT_DIR := ./cmd/my-operator-tests
+#   TESTS_EXT_OUTPUT_DIR := ./cmd/my-operator-tests  # Optional, defaults to TESTS_EXT_DIR
+#
+# Required variables:
+#   TESTS_EXT_BINARY - Name of the test extension binary
+#   TESTS_EXT_DIR - Directory containing test extension code (e.g., ./cmd/my-operator-tests)
+#
+# Optional variables:
+#   TESTS_EXT_OUTPUT_DIR - Output directory for the binary (defaults to TESTS_EXT_DIR)
+
+# Default values
+TESTS_EXT_OUTPUT_DIR ?= $(TESTS_EXT_DIR)
+
+# -------------------------------------------------------------------
+# Ensure test binary has correct name and location
+# -------------------------------------------------------------------
+.PHONY: tests-ext-build
+tests-ext-build: build
+	@mkdir -p $(TESTS_EXT_OUTPUT_DIR)
+	@# Try to find and move the binary - check both the desired name and the package-based name
+	@if [ -f $(TESTS_EXT_BINARY) ] && [ ! -f $(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY) ]; then \
+		mv $(TESTS_EXT_BINARY) $(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY); \
+	elif [ -f $(shell basename $(TESTS_EXT_DIR)) ] && [ ! -f $(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY) ]; then \
+		mv $(shell basename $(TESTS_EXT_DIR)) $(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY); \
+	fi
+
+# -------------------------------------------------------------------
+# Update test metadata (optional - only if metadata directory exists)
+# -------------------------------------------------------------------
+.PHONY: tests-ext-update
+tests-ext-update: tests-ext-build
+	@if [ -d "$(TESTS_EXT_DIR)/.openshift-tests-extension" ]; then \
+		$(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY) update; \
+	else \
+		echo "Note: No metadata directory found at $(TESTS_EXT_DIR)/.openshift-tests-extension - skipping update"; \
+	fi
+
+# -------------------------------------------------------------------
+# Verify test metadata is up to date (optional - only if metadata directory exists)
+# -------------------------------------------------------------------
+.PHONY: verify-tests-ext-metadata
+verify-tests-ext-metadata: tests-ext-update
+	@if [ -d "$(TESTS_EXT_DIR)/.openshift-tests-extension" ]; then \
+		if ! git diff --exit-code $(TESTS_EXT_DIR)/.openshift-tests-extension; then \
+			echo "ERROR: Metadata is out of date. Please run 'make tests-ext-update' and commit the result."; \
+			exit 1; \
+		fi; \
+	else \
+		echo "Note: No metadata directory found - skipping verification"; \
+	fi
+
+# -------------------------------------------------------------------
+# List available test names
+# -------------------------------------------------------------------
+.PHONY: list-test-names
+list-test-names: tests-ext-build
+	@$(TESTS_EXT_OUTPUT_DIR)/$(TESTS_EXT_BINARY) list -o names
+
+# -------------------------------------------------------------------
+# Run test suite
+# -------------------------------------------------------------------
+.PHONY: run-suite
+run-suite: tests-ext-build
```

Comment:
let's start simple and define only the target we discussed which is `run-suite`

Reply:
@p0lyn0mial only keeping 3 
  1. tests-ext-build - Builds the test binary
  2. run-suite - Runs test suites
  3. tests-ext-clean and clean - Cleans the test binary

Reply:
@p0lyn0mial 1 needed because it required in Dockerfile because in Dockerfile we only build, run-suite  only required to run suite.
removed 3rd one.

Reply:
@gangwgr try removing `make tests-ext-build` from the docker file and see if you get the binary. I think `make build` does the trick - just the path to the binary might different.